### DISCUSSION TITLE
Is termux compatible at all?

### DISCUSSION THREAD

**Discussion by framgena** [Beastwick18/nyaa#33] [Q&A] [answered]
I'm totally noob at all this, just wanted to know: is it possible to run on android with termux?:) or not?

It gives errors connected with rust in return, or as result of script as I see.

**Comment by Beastwick18** [comment]
Termux support _should_ be working, which  was resolved in #27. You'd have to build from source, so you'd run something like:
```sh
git clone https://github.com/Beastwick18/nyaa
cd nyaa
cargo install --path .
```
which requires both `git` and `cargo` be installed.

And make sure to add to your `~/.bashrc` file:
```sh
export PATH="$HOME/.cargo/bin:$PATH"
```

So that it can find it once installed. You should then be able to run `nyaa` after restarting your terminal.

**Comment by framgena** [comment]
Much thanks, but still hasn't working:)

Double checked "git" and"cargo" just in case - everything installed.
What exactly must I do? Step by step preferably.
CD nyaa then "cargo install" and add this 
"export PATH="$HOME”... etc"  ?

It gives error even after "cargo path". This one

 $ cargo install --path
error: a value is required for '--path <PATH>' but none was supplied

**Comment by Beastwick18** [comment]
The first command, `git clone https://github.com/Beastwick18/nyaa` copies the contents of this repo into the folder nyaa so it can be run.

Then you have to move into that directory to build the app, so `cd nyaa`

The last command, `cargo install --path .` (make sure to include the period) builds the app and installs it. However, it installs it to a place your terminal won't be able to find by default. So, we have to edit the configuration for the terminal to find it.

That's when we add `export PATH="$HOME/.cargo/bin:$PATH"` to `~/.bashrc`. This makes programs installed by cargo visible to your terminal. You can open up this file for editing by running the command `nano ~/.bashrc` then typing out that `export ...` line somewhere towards the end of the file.

Then you should be able to run the command `nyaa` :)

**Reply by framgena** [reply]
Much obliged for your help!
I didn't pay much attention to "." at line with "cargo":D
That's the only reason of my failure.
I managed to start it up finally!

But after closing termux app and reopening it -"nyaa" and as "CD nyaa" gives me nothing, so I'm not sure if its installed already to termux "data/data" folder.

Could you please tell me what exactly file in "nano editor" from "nyaa" folder I must edit with these lines? It's "cargo.toml"? I ended with this


.cargo                 (dir)
.config                (dir)
.ssh                   (dir)
.termux                (dir)
nyaa                   (dir)
rutracker-master       (dir)
storage                (dir)
.bash_history         981  B
.bashrc Cargo.toml      2 KB
.bashrc.save           67  B

This "bashrc cargo.toml" is a file I edited.

**Reply by Beastwick18** [reply, marked as answer]
> This "bashrc cargo.toml" is a file I edited.

The file should be named `.bashrc`, without the `cargo.toml` part. Adding `export PATH="$HOME/.cargo/bin:$PATH"` to it will make it so `nyaa` will be found by the terminal.

So you could run `nano ~/.bashrc`, then add `export PATH="$HOME/.cargo/bin:$PATH"` to the file towards the end, then press <kbd>Ctrl</kbd>+<kbd>X</kbd> to close nano, and press <kbd>y</kbd> when it asks if you want to save your changes. Restarting your terminal should reload the `.bashrc` file.

**Reply by framgena** [reply]
Damn, it was that easy...:) I just already figured out your "nano -/ bashrc" opens this file directly in nano explorer:D and I must just add that line and overwrite. That's all. Never used "nano" before so it's "clean screen" while file is opened confused me and I decided something was going wrong.

Now nyaa opens flawlessly. Deal with proxy is all what's left to figure out "how". But that's much easier.

Thanks a ton for your patience and help for a total noob like me, Steve! Best wishes and have a nice day!

**Comment by framgena** [comment]
Good day, Steve!
I'm sorry for disturbance again, but one last question, OK?:)
Is it possible to download directly, like webtor do, for example?
I tried every option in "d", except  obvious torrent clients, but nothing
downloaded through termux directly or through inner phone "downloads" for
example.

If it's not possible at all (don't have access to WiFi downloads right now)
- well,... :)

пт, 22 нояб. 2024 г. в 08:32, Steven Culwell ***@***.***>:

**Reply by Beastwick18** [reply]
## Option 1
If you're fine with downloading directly to your phone, you can just install a torrent client app like `Libretorrent`.

You'll want to modify the config for nyaa to open magnet links instead of torrent links. You can open the config in nano with:
```sh
nano ~/.config/nyaa/config.toml
```
and look for the line `[client.default_app]` and modify the line under it from `use_magnet = false` to `use_magnet = true`. Save the and close the file with <kbd>Ctrl</kbd>+<kbd>X</kbd> and `y` to save changes.

Then you can open `nyaa` and switch the download client to `Default App` in nyaa by pressing <kbd>d</kbd> and navigating to it and pressing <kbd>Enter</kbd>. Now, any time you press enter on any of the torrents, it will open in Libretorrent and start downloading. Once it finishes downloading the torrent, you can open the file.

## Option 2
If you would rather open in webtor, the easiest way I can think of doing this would be to modify the `Run Command` download client. We'll modify it to first copy the torrents magnet link to the clipboard, and then open `webtor.io` where you can paste the magnet link.

First, you'll have to install `termux-api` so we can modify the clipboard. You can install it by running the command:
```sh
pkg install termux-api
```

To modify `Run Command`, you'll have to open the file `~/.config/nyaa/config.toml`, which you can open in nano by running:
```
nano ~/.config/nyaa/config.toml
```

You'll want to find a line under `[client.command]` that looks like `command="..."`. You can change it to something like:
```toml
command='termux-clipboard-set "{magnet}" && termux-open-url "https://webtor.io"'
```
(Note that it's `''` surrounding the command. This allows `""` to be used inside more easily)

Now you can open `nyaa`, and change the download client to `Run Command` by pressing <kbd>d</kbd> and navigating to it and pressing <kbd>Enter</kbd>. Now, any time you press enter on any of the torrents, it will copy the magnet link to your clipboard, and open `webtor.io` in your browser. Paste the link into webtor and all should work!

**Comment by framgena** [comment]
Oh... I'm sorry my request makes you write all this detailed, Steve! I
hoped for a basic answer:)
I got it, so now this application is much convenient way to find and
download torrents but not allows to download via 4g through app exactly.
OK, I'll continue to use webtor then:)

Thank you very much  for all your help, have a nice day!

сб, 23 нояб. 2024 г. в 19:14, Steven Culwell ***@***.***>:
